src/mumble/GlobalShortcut_win.cpp

```diff
@@ -229,20 +241,25 @@ void GlobalShortcutWin::injectKeyboardMessage(MSG *msg) {
 		case WM_SYSKEYUP:
 			break;
 		default:
-			return;
+			return false;
 	}
 
 	DWORD scancode = (msg->lParam >> 16) & 0xff;
 	DWORD vkcode = msg->wParam;
 	bool extended = !!(msg->lParam & 0x01000000);
 	bool up = !!(msg->lParam & 0x80000000);
 
-	qApp->postEvent(this, new InjectKeyboardMessageEvent(scancode, vkcode, extended, !up));
+	std::promise<bool> suppressionPromise;
+	std::future<bool> suppressionFuture = suppressionPromise.get_future();
```

Comment:
I think I would prefer a design where the event owns the promise from the start and we ask it for the future before loosing ownership. No need to even have the promise "out here" only to move it in. E.g.
```cpp
InjectKeyboardMessageEvent *ikme = new InjectKeyboardMessageEvent(scancode, vkcode, extended, !up);
std::future<bool> shouldSuppress = ikme->shouldSuppress();
qApp->postEvent(this, ikme); // Yields ikme ownership
return shouldSuppress.get();
```
or something like that.